docs/setup/glb-healthcheck-configuration.md

```diff
@@ -0,0 +1,48 @@
+# glb-healthcheck configuration
+
+The `glb-healthcheck` component is the application that sends probes to determine liveliness of the 
+L7 back-ends for which `glb-director` is acting as an L4 load-balancer.
+
+
+`glb-healthcheck` does the health-checking for those L7 back-ends which are listed in the configuration for `glb-director`.
+ It processes two json files: 
+* A source JSON file: to learn the identities of the L7 back-ends whose health is to be checked. This file also lists the protocol to use and the specific protocol-parameters to use when crafting the health-check probes. By default this file is present at `/etc/glb/forwarding_table.src.json`.
+* A destination JSON file: to which `glb-healthcheck` writes the contents of this file along with health-status for each of the L7 back-ends listed in the `source JSON file`. The `glb-director CLI` converts this json file into a `.bin` file which serves as the configuration file to be read by `glb-director`. 
+By default this file is present at `/etc/glb/forwarding_table.checked.json`.
+  
+The configuration for `glb-healthcheck` component is present by default at:
+* `/etc/glb/healthcheck.conf`: This file lists the locations of the 2 json files processed by `glb-healthcheck`.
+* `/etc/glb/forwarding_table.src.json`: This file provides the configuration for `glb-director` if health-checking of the L7 back-ends is not to be performed. It also provides configuration for `glb-healthcheck` for the actual sending of the probes.
+
+## `/etc/glb/healthcheck.conf`
+
+### forwarding_table
+Under `src` and `dst`: configuration for the file-path for the source and destination JSON files to be processed by `glb-healthcheck`.
+By default, `src` is `/etc/glb/forwarding_table.src.json`; and `dst` is `/etc/glb/forwarding_table.checked.json` .
+
+### reload_command
+Specifies the commands to run upon creation, by `glb-healthcheck`, of a new/updated destination JSON.
+ 
+## `/etc/glb/forwarding_table.src.json`
```

Comment:
I would have expected to configure the healthcheck configuration, at least in a global way, in `/etc/glb/healthcheck.conf` instead of in the forwarding table file, which is more about the binds and backends than how the healthchecker happens to test them. Was there a particular reason for putting it in the forwarding table file?

Reply:
I started out wanting to make this configurable on a per back-end basis to factor for different kinds of hardware; or even on a per-table basis to factor for different kinds of per-service loads on back-ends.
However, doing that would have required a bit bigger architectural changes given the simplicity of the code in doHealthCheckRound().  So, postponed that for a bit.
The per-table and per-backend params would appear to belong to the forwarding_table JSONs. 
With an implementation for per-backend/table params, having a global setting for the same param in the same file makes things simpler.

Reply:
:+1: That makes sense, thanks for the explanation!